commons/src/scanner/symbol.rs

```diff
@@ -176,7 +176,10 @@ impl Symbol<'_> {
     /// It is assumed (and checked in debug release) that the symbols are in contiguous order.
     ///
     /// Returns `None` if the referenced input is not same in all symbols.
-    pub fn flatten_iter<'s>(mut iter: impl Iterator<Item = &'s Symbol<'s>>) -> Option<&'s str> {
+    pub fn flatten_iter<'o, 's>(mut iter: impl Iterator<Item = &'o Symbol<'s>>) -> Option<&'s str>
```

Comment:
Good observation. I will try to rename the lifetimes so that naming is more consistent. Also, I will document them in doc comments, as that should help with understanding (and works nice in IDE hover pop-up documentation etc).